### PR TITLE
avoid bool conversion

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/property_events.js [hunk 1]
```diff
@@ -51,10 +51,10 @@ function propertyWillChange(obj, keyName, _meta) {
 
   let watching = meta && meta.peekWatching(keyName) > 0;
   let possibleDesc = obj[keyName];
-  let desc = (possibleDesc !== null && typeof possibleDesc === 'object' && possibleDesc.isDescriptor) ? possibleDesc : undefined;
+  let isDescriptor = possibleDesc !== null && typeof possibleDesc === 'object' && possibleDesc.isDescriptor;
 
-  if (desc && desc.willChange) {
```
Comment (bekzod) [inline packages/ember-metal/lib/property_events.js]: bool conversion as `desc` is either `obj` or `undefined `

File: packages/ember-metal/lib/property_events.js [hunk 2]
```diff
@@ -90,11 +90,11 @@ function propertyDidChange(obj, keyName, _meta) {
   }
 
   let possibleDesc = obj[keyName];
-  let desc = (possibleDesc !== null && typeof possibleDesc === 'object' && possibleDesc.isDescriptor) ? possibleDesc : undefined;
+  let isDescriptor = possibleDesc !== null && typeof possibleDesc === 'object' && possibleDesc.isDescriptor;
 
   // shouldn't this mean that we're watching this key?
-  if (desc && desc.didChange) {
```
Comment (bekzod) [inline packages/ember-metal/lib/property_events.js]: bool conversion as `desc` is either `obj` or `undefined `

File: packages/ember-metal/lib/property_events.js [hunk 3]
```diff
@@ -175,9 +175,9 @@ function iterDeps(method, obj, depKey, seen, meta) {
     if (!value) { return; }
 
     possibleDesc = obj[key];
-    desc = (possibleDesc !== null && typeof possibleDesc === 'object' && possibleDesc.isDescriptor) ? possibleDesc : undefined;
+    isDescriptor = possibleDesc !== null && typeof possibleDesc === 'object' && possibleDesc.isDescriptor;
 
-    if (desc && desc._suspended === obj) {
```
Comment (bekzod) [inline packages/ember-metal/lib/property_events.js]: bool conversion as `desc` is either `obj` or `undefined `

File: packages/ember-runtime/lib/system/core_object.js [hunk 4]
```diff
@@ -146,8 +143,11 @@ function makeCtor() {
               value = assign({}, originalValue, value);
             }
 
-            if (desc) {
```
Comment (bekzod) [inline packages/ember-runtime/lib/system/core_object.js]: bool conversion as `desc` is either `obj` or `undefined `

**Review by stefanpenner — APPROVED** [review]

**Pull request merged**